frontend/src/components/Login.tsx

```diff
@@ -0,0 +1,164 @@
+'use client';
+import { useAuthStore, useUserStore } from '../context/authStore';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { FormEvent, useState } from 'react';
+import {
+  QueryClient,
+  QueryClientProvider,
+  useMutation
+} from '@tanstack/react-query'
+
+const queryClient = new QueryClient();
+
+export default function Login() {
+  return (
+    <QueryClientProvider client={queryClient}>
+      <LoginForm />
+    </QueryClientProvider>
+  );
+}
+
+function LoginForm() {
+  const [formData, setFormData] = useState({ email: '', password: '' });
+  const [error, setError] = useState('');
```

Comment:
error & loading states could be eliminated using react query's states
- https://tanstack.com/query/v3/docs/framework/react/guides/mutations
- https://refine.dev/blog/react-query-guide/#mutating-data